src/network/live_configuration/update_thread_worker.rs

```diff
@@ -103,14 +105,24 @@ impl<T: ServerClient> UpdateThreadWorker<T> {
     fn update_configuration_from_server_and_current_mode(&self) -> Result<()> {
         match self.server_client.get_configuration(&self.configuration_id) {
             Ok(config) => {
-                *self.configuration.lock()? = Some(config);
-                *self.current_mode.lock()? = CurrentMode::Online;
+                let mut current_config = self.configuration.lock()?;
+                *current_config = Some(config);
+
+                self.current_mode
+                    .set(CurrentMode::Online)
+                    .map_err(|_| Error::CannotAcquireLock)?;
```

Comment:
We already provide a `Into` implementation from the `PoisonError` to our `Error::CannotAcquireLock`? We don't need this `map_err` here


```suggestion
                self.current_mode
                    .set(CurrentMode::Online)?;
```